jsonschema/validators.py

```diff
@@ -595,9 +598,38 @@ def from_schema(
 
         return cls(base_uri=id_of(schema), referrer=schema, *args, **kwargs)
 
+    def http_handler(self, uri):
+        try:
+            import requests
+        except ImportError:
+            pass
+        else:
+            if hasattr(requests.Response, "json"):
+                session = requests.Session()
+
+                requests_supports_scheme = True
+                if urisplit(uri).scheme == "file":
+                    try:
+                        import requests_file
```

Comment:
This library doesn't seem too widely used. Is it actually?

Reply:
Not particularly, but in all it's quite simple (but not simple enough to warrant copying the source imo). Off the top of my head, urllib2 has a builtin file schema handler. There are several options to this:

1. Remove requests file and re-write it (see https://stackoverflow.com/a/22989322/1215241)
1. Don't use requests for files, just use urllib2 / urllib.request
1. Don't use any http getter and manually open the file with open() (see https://github.com/python/cpython/blob/3.7/Lib/urllib/request.py#L1976)

I avoided the third option because my knowledge of the path handling quirks is limited. Feel free to make a suggestion here.